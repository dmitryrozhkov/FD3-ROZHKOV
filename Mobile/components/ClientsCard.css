.clientsCard {
    margin: 10px auto;      
    height: 200px;
    width: 200px;
    background: rgb(218, 195, 166);
    border: 1px solid blue;
    text-align: left;
    padding: 10px 0 0 10px;
}
