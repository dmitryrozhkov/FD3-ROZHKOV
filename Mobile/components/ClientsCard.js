import React from 'react';

import {clientsEvents} from './events';

import './ClientsCard.css';


  class ClientsCard extends React.PureComponent {    
    
    saveFio = () => {
        if ( this.newFioRef&&this.newBalanceRef&&this.newIdRef ) {
           let newId=parseInt(this.newIdRef.value);
           let newFio=this.newFioRef.value;  
           let newBalance=parseInt(this.newBalanceRef.value);   
         
        clientsEvents.emit('EAddFioClicked', newId, newFio, newBalance);
     }       
    }

    deleteFio = () => {
        if ( this.deleteIdRef) {
           let deleteId=parseInt(this.deleteIdRef.value);        
      
        clientsEvents.emit('EDeleteFioClicked', deleteId);
     }       
    }

    newIdRef = null;
    setNewIdRef = (ref) => {
        this.newIdRef=ref;       
    }

    newFioRef = null;
    setNewFioRef = (ref) => {
        this.newFioRef=ref;        
    }

    newBalanceRef = null;
    setNewBalanceRef = (ref) => {
        this.newBalanceRef=ref;    
    }

    deleteIdRef = null;
    setDeleteIdRef = (ref) => {
        this.deleteIdRef=ref;       
    }


    render () {
        console.log("ClientsCard render");

        if (this.props.cardWorkMode==1) {     
            return (                  
                <div className = 'clientsCard'>
                          <label>ID:</label><br/> 
                          <input type='text' ref={this.setNewIdRef} /><br/><br/> 
                          <label>Фамилия:</label><br/>                                 
                          <input type='text' ref={this.setNewFioRef} /><br/><br/>
                          <label>Баланс:</label><br/>                                 
                          <input type='text' ref={this.setNewBalanceRef} /><br/><br/>
                          <button onClick={this.saveFio}>Сохранить</button>
                                                                                             
                </div>
        )
    }
        if (this.props.cardWorkMode==2) {     
            return (                  
                <div className = 'clientsCard'>
                          <label>ID:</label><br/> 
                          <input type='text' ref={this.setDeleteIdRef} /><br/><br/>                         
                          <button onClick={this.deleteFio}>Удалить</button>
                                                                                             
                </div>
        )
    }
    else return null
      }
    }       
  
      export default ClientsCard ;
  
  