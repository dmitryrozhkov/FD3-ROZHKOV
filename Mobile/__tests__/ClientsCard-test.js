"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import ClientsCard from '../components/ClientsCard';

test('работа ClientsCard', () => {

let counter= [1, 2, 3, 4]

  counter.forEach( (n) =>     {     
    
  const component = renderer.create(
    <ClientsCard cardWorkMode={n} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  }); 
    
});