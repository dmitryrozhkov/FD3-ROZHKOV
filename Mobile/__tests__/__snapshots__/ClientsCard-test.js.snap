// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа ClientsCard 1`] = `
<div
  className="clientsCard"
>
  <label>
    ID:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <label>
    Фамилия:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <label>
    Баланс:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Сохранить
  </button>
</div>
`;

exports[`работа ClientsCard 2`] = `
<div
  className="clientsCard"
>
  <label>
    ID:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Удалить
  </button>
</div>
`;

exports[`работа ClientsCard 3`] = `
<div
  className="clientsCard"
>
  <label>
    ID:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <label>
    Новая фамилия:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Изменить
  </button>
</div>
`;

exports[`работа ClientsCard 4`] = `
<div
  className="clientsCard"
>
  <label>
    ID:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <label>
    Новый баланс:
  </label>
  <br />
  <input
    type="text"
  />
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Изменить
  </button>
</div>
`;
